baggage/baggage.go

```diff
@@ -630,6 +631,91 @@ func skipSpace(s string, offset int) int {
 	return i
 }
 
+var safeKeyCharset = [utf8.RuneSelf]bool{
```

Comment:
Have you compared of there will be the performance improvement if we change the type to `[utf8.RuneSelf]struct{}` and use the `_, ok := safeKeyCharset[s]` idiom?

Reply:
Got a lot worse with a `map[byte]struct{}`. 

```
$ benchstat /tmp/main.txt /tmp/struct-variant.txt
name                             old time/op    new time/op    delta
New-8                               905ns ±11%    1136ns ± 8%   +25.46%  (p=0.000 n=19+19)
NewMemberRaw-8                     13.1ns ± 5%    43.4ns ±17%  +230.30%  (p=0.000 n=17+19)
Parse-8                             758ns ±10%    1594ns ± 6%  +110.14%  (p=0.000 n=20+16)
String-8                           1.33µs ±11%    4.52µs ±14%  +238.77%  (p=0.000 n=18+20)
ValueEscape/nothing_to_escape-8    7.97ns ± 4%   54.75ns ± 4%  +587.22%  (p=0.000 n=19+17)
ValueEscape/requires_escaping-8    31.1ns ± 4%    83.7ns ± 7%  +169.38%  (p=0.000 n=18+18)
ValueEscape/long_value-8            758ns ± 5%    5503ns ± 9%  +625.99%  (p=0.000 n=19+20)

name                             old alloc/op   new alloc/op   delta
New-8                                832B ± 0%      832B ± 0%      ~     (all equal)
NewMemberRaw-8                      0.00B          0.00B           ~     (all equal)
Parse-8                              832B ± 0%      832B ± 0%      ~     (all equal)
String-8                           1.06kB ± 0%    1.06kB ± 0%      ~     (all equal)
ValueEscape/nothing_to_escape-8     0.00B          0.00B           ~     (all equal)
ValueEscape/requires_escaping-8     16.0B ± 0%     16.0B ± 0%      ~     (all equal)
ValueEscape/long_value-8             576B ± 0%      576B ± 0%      ~     (all equal)

name                             old allocs/op  new allocs/op  delta
New-8                                16.0 ± 0%      16.0 ± 0%      ~     (all equal)
NewMemberRaw-8                       0.00           0.00           ~     (all equal)
Parse-8                              7.00 ± 0%      7.00 ± 0%      ~     (all equal)
String-8                             18.0 ± 0%      18.0 ± 0%      ~     (all equal)
ValueEscape/nothing_to_escape-8      0.00           0.00           ~     (all equal)
ValueEscape/requires_escaping-8      1.00 ± 0%      1.00 ± 0%      ~     (all equal)
ValueEscape/long_value-8             2.00 ± 0%      2.00 ± 0%      ~     (all equal)
```